java-client/src/main/java/co/elastic/clients/transport/ElasticsearchTransportBase.java

```diff
@@ -377,28 +379,42 @@ private <ResponseT> ResponseT decodeTransportResponse(
     ) throws IOException {
 
         if (endpoint instanceof JsonEndpoint) {
+
+            // Expecting a body
+            if (entity == null) {
+                throw new TransportException(
+                    clientResp,
+                    "Expecting a response body, but none was sent",
+                    endpoint.id()
+                );
+            }
+            InputStream content = entity.asInputStream();
+            InputStream contentForException = null;
+
+            // if the option to print the original body has been set, the body has to be
+            // copied first to another stream to be read again by the exception class
+            if(options().retrieveOriginalJsonResponseOnException()) {
```

Comment:
We already have something similar for error responses: https://github.com/elastic/elasticsearch-java/blob/5d466ade855d62900150f52791abf3ef28bec6bf/java-client/src/main/java/co/elastic/clients/transport/ElasticsearchTransportBase.java#L336-L339

We could do the same here. Or even factorize it in a new method `BinaryData ensureRepeatable()` in `BinaryData`